product_variant_weight/models/product.py

```diff
@@ -0,0 +1,56 @@
+# -*- coding: utf-8 -*-
+# © 2016 Sergio Teruel <sergio.teruel@tecnativa.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from openerp import api, fields, models
+from openerp.addons import decimal_precision as dp
+
+
+class ProductTemplate(models.Model):
+    _inherit = 'product.template'
+
+    def _prepare_weight_values(self, vals):
+        values = {}
+        if 'weight' in vals:
+            values['weight'] = vals['weight']
+        if 'weight_net' in vals:
+            values['weight_net'] = vals['weight_net']
+        return values
+
+    @api.multi
+    def write(self, vals):
+        res = super(ProductTemplate, self).write(vals)
+        for product in self:
+            values = self._prepare_weight_values(vals)
```

Comment:
Given you are doing exactly the same in `write` and `create`, I think `_prepare_weight_values` would be better as `_propagate_weight_to_variants`, and should have this logic you write from lines 24 to 26. Just for DRY.